control_protocol.md

```diff
@@ -330,17 +330,490 @@ A Component MUST also offer a list of all possibly callable methods in accordanc
 For such a RPC message, the first content frame MUST consist in a JSON-RPC compatible content, for example a single request object or a batch of request objects.
 
 
+### List of methods
+
+List of methods defined by LECO.
+You MAY implement the optional methods of this list and you MUST implement the methods obligatory for your type of Component.
+All methods implemented in a Component MUST adhere to this list or MUST have a name different to any method in this list.
+
+
+#### Component
+
+Any Component, i.e. any participant in the LECO protocol, MUST offer the following methods.
+
+::: json
```

Comment:
Do we really want to put json into the markdown documents? Shouldn't we rather put the json schema file somewhere (for automated use) and either parse (automated) or paraphrase (manual)/present it in text?

Reply:
As we have a standard, which defines, how to define RPC calls, we should use it and not some other method (e.g. python method definitions).

You're right, the JSON text is quite large in comparison to python definitions or human text.

I'm against manual rephrasing (although we might not change it that often), so it should be an automatic process.

What do you think about importing name and description (which we should add), such that the documentation will look like this:

"""
A Component may implement the following methods, for full specification see `link_to_json_file`:

- `shut_down`: Shut down the Component.
- `set_log_level`: Set the Component's logging level.

"""

Reply:
I see your point regarding not manually duplicating stuff.

The specification document should be complete-ish, asking a reader to manually crawl through a json file elsewhere is quite uncomfortable. Maybe we can link/include specific sections of a json file using Sphinx?
A quick search yielded https://sphinx-jsonschema.readthedocs.io/en/latest/directive.html#an-example which looks like what we want?

Reply:
nice!